hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/AbstractEmbeddableMapping.java

```diff
@@ -91,8 +92,12 @@ public Object[] getValues(Object compositeInstance) {
 			return optimizer.getAccessOptimizer().getPropertyValues( compositeInstance );
```

Comment:
The generated access optimizer implementation should also be adapted to add the class name to the end if we decide to do this.

Reply:
Will be done as part of https://hibernate.atlassian.net/browse/HHH-18110